meson.build

```diff
@@ -0,0 +1,533 @@
+        cfg_var.set('HAVE_JPEG_EXT_PARAMS', '1')
+    endif
+endif
+
+# Look for libspng first
+# 0.6.1 uses "libspng.pc", git master libspng uses "spng.pc"
+# We can simplify this when requiring meson>=0.60.0
+spng_dep = dependency('libspng', version: '>=0.6', required: false)
+if not spng_dep.found()
+    spng_dep = dependency('spng', version: '>=0.6', required: get_option('spng'))
+endif
+if not get_option('spng').disabled() and spng_dep.found()
+    libvips_deps += spng_dep
+    cfg_var.set('HAVE_SPNG', '1')
+endif
+
+# we can have both PNG and SPNG enabled, with SPNG for read and PNG for write
+png_dep = dependency('libpng', version: '>=1.2.9', required: get_option('png'))
+if png_dep.found()
+    libvips_deps += png_dep
+    cfg_var.set('HAVE_PNG', '1')
+    if cc.has_function('png_set_chunk_malloc_max', prefix: '#include <png.h>', dependencies: png_dep)
+        cfg_var.set('HAVE_PNG_SET_CHUNK_MALLOC_MAX', '1')
+    endif
+endif
+
+
+# libwebp ... target 0.6+ to reduce complication
+# webp has the stuff for handling metadata in two separate libraries  -- we
+# insit on having all of them
+libwebp_dep = dependency('libwebp', version: '>=0.6', required: get_option('png'))
+if libwebp_dep.found()
+    libvips_deps += libwebp_dep
+    libvips_deps += dependency('libwebpmux', version: '>=0.6')
+    libvips_deps += dependency('libwebpdemux', version: '>=0.6')
+    cfg_var.set('HAVE_LIBWEBP', '1')
+endif
+
+pangocairo_dep = dependency('pangocairo', required: get_option('pangocairo'))
+if pangocairo_dep.found()
+    libvips_deps += pangocairo_dep
+    cfg_var.set('HAVE_PANGOCAIRO', '1')
+endif
+
+fontconfig_dep = dependency('fontconfig', required: get_option('fontconfig'))
+if fontconfig_dep.found() and pangocairo_dep.found()
+    libvips_deps += fontconfig_dep
+    cfg_var.set('HAVE_FONTCONFIG', '1')
+endif
+
+libtiff_dep = dependency('libtiff-4', required: get_option('tiff'))
+if libtiff_dep.found()
+    libvips_deps += libtiff_dep
+    cfg_var.set('HAVE_TIFF', '1')
+    # ZSTD and WEBP in TIFF added in libtiff 4.0.10
+    if cc.get_define('COMPRESSION_WEBP', prefix: '#include <tiff.h>', dependencies: libtiff_dep) != ''
+        cfg_var.set('HAVE_TIFF_COMPRESSION_WEBP', '1')
+    endif
+endif
+
+# 2.40.3 so we get the UNLIMITED open flag
+librsvg_dep = dependency('librsvg-2.0', version: '>=2.40.3', required: get_option('rsvg'))
+cairo_dep = dependency('cairo', version: '>=1.2', required: get_option('rsvg'))
+if librsvg_dep.found() and cairo_dep.found()
+    libvips_deps += librsvg_dep
+    libvips_deps += cairo_dep
+    cfg_var.set('HAVE_RSVG', '1')
+    # 2.46 for rsvg_handle_render_document
+    if librsvg_dep.version().version_compare('>=2.46')
+        cfg_var.set('HAVE_RSVG_HANDLE_RENDER_DOCUMENT', '1')
+    endif
+endif
+
+zlib_dep = dependency('zlib', version: '>=0.4', required: get_option('zlib'))
+if zlib_dep.found()
+    libvips_deps += zlib_dep
+    cfg_var.set('HAVE_ZLIB', '1')
+endif
```

Comment:
This can be merged into the if statement above. Since libvips only uses zlib to handle svgz images.